lib/isCurrency.js

```diff
@@ -18,7 +18,7 @@ function currencyRegex(options) {
   });
   var symbol = "(\\".concat(options.symbol.replace(/\./g, '\\.'), ")").concat(options.require_symbol ? '' : '?'),
       negative = '-?',
-      whole_dollar_amount_without_sep = '[1-9]\\d*',
+      whole_dollar_amount_without_sep = options.require_thousands_separator ? '' : '[1-9]\\d*',
```

Comment:
Could you please clarify this line? The var here is `_without_sep`, while we are also checking `options.require_thousands_sep`, are these two separate _separators_?

Reply:
the same separator, a comma in the default settings. The code intent isn't clear, and the empty string passed in becomes unnecessary overhead. can fix both with the addition of whole_dollar_amount_without_sep to valid_whole_dollar_amounts happen optionally.  will commit to pr.
```suggestion
    whole_dollar_amount_without_sep = '[1-9]\\d*',
    whole_dollar_amount_with_sep = `[1-9]\\d{0,2}(\\${options.thousands_separator}\\d{3})*`,
    valid_whole_dollar_amounts = ['0', whole_dollar_amount_with_sep];
  if (!options.require_thousands_separator) {
    valid_whole_dollar_amounts.push(whole_dollar_amount_without_sep);
  }
  const whole_dollar_amount = `(${valid_whole_dollar_amounts.join('|')})?`,

```